### ISSUE TITLE
lock screen disables mouse and keyboard sharing

### ISSUE THREAD

**Issue by nbolton** [synergy/synergy#3325]
**Imported issue:**

* Author: Tony Pan
* Date: 2012-10-10 18:12:05
* Legacy ID: 3395

**Steps to reproduce:**

1. install synergy 1.4.10 on ubuntu 12.04 via Ubuntu Software Center
2. configured synergy server to run before login on lightdm, per https://help.ubuntu.com/community/SynergyHowto.  restart ubuntu system
3. install synergy 1.4.10 on windows 7 enterprise
4. run synergy as client on windows 7 enterprise.
5. make sure the 2 systems connect and am able to use ubuntu keyboard and mouse (should work before log-in or after log-in)
6. either wait for ubuntu to lock screen, or manually lock screen on ubuntu

expected: mouse and keyboard continue to be shared from ubuntu to windows

actual: ubuntu mouse and keyboard are now restricted to the locked ubuntu screen.  after unlock, behavior returns to normal.

**Versions and operating systems:**

synergy 1.4.10
ubuntu 12.04
windows 7 enterprise

**Temporary workarounds:**

no

**Additional comments:**

great software.

**Comment by dragon788** [comment]
@nbolton Is this a wontfix because its due to a limitation in Ubuntu itself, or does a newer version of Synergy correct this? I'm wobbling between gnome-screensaver and Xscreensaver to try and work around this issue, but that doesn't seem to fix it either.

**Comment by unilaterus** [comment]
I am experiencing the same or very similar issue:

Setup:
Ubuntu Server
Windows 7 client

When Windows 7 Client locks screen the synergy immediately disconnects and then the client reconnects after some short amount of time (5000ms) it also (since client disconnected) jumps the keyboard/mouse from client to server.

So in my case I can still control the Windows client but only after waiting a full 5 seconds for the server to find the client again.

**Comment by dragon788** [comment]
@unilaterus I'm guessing your Windows machine is on a domain since it is the Enterprise version? I've run into some strangeness where the service that Synergy uses works great as long as I'm logged in, but when the screen locks due to Group Policy, I am unable to access the machine via Synergy, even though it detects that there is a screen, the policy triggering backgrounds Synergy to not have access to the Ctrl+Alt+Del or foreground lock. I haven't tested with the very latest versions yet, but it was pretty consistent for a while so I have a small USB 2 port switch on my desk to throw my keyboard back and forth for login.

**Comment by unilaterus** [comment]
@dragon788 mine is on a domain but I have already fixed the policy so that i can Ctrl+Alt+Del via synergy so that part isn't a particular problem.

My issue is that for 5 seconds after locking or unlocking the windows machine I cannot access it via synergy. After 5 seconds I can drag mouse over and type normally I just can't for 5 seconds after lock or unlock.

**Comment by liyam** [comment]
I have a similar issue, too.

Synergy Version: 1.7.5
Client: Windows 7
Server: Centos 7

I have synergy autostart configuration for gdm.

However, during the gdm login screen or locked screen from the server side, the keyboard and the mouse are only restricted to access only the server screen.
I can't access the client screen even though the client logs show that it is connected to the server.

I hope to get any tip or patch for fixing it soon.

**Comment by pgargan** [comment]
Similar issue, very annoying.

Synergy Pro 1.8.7
Ubuntu 12.04 server
Windows 10 Enterprise client
Ctrl+Alt+Del enabled for unlock as per https://github.com/symless/synergy/wiki/User-FAQ#Why_doesnt_CtrlAltDel_work_on_secondary_screens

When the Win 10 client locks, the connection drops for ~5 seconds. After that, I can move the mouse back onto the client and send Ctrl+Alt_Del. But immediately when I do this, the connection drops again, meaning I often inadvertently start typing my password into whatever app is open on the server. 

Server log shows entries like:

```
[2017-03-27T17:10:29] NOTE: client "xxxxx" has disconnected
[2017-03-27T17:10:34] NOTE: accepted client connection
[2017-03-27T17:10:34] NOTE: client "xxxxx" has connected
[2017-03-27T17:10:37] NOTE: client "xxxxx" has disconnected
[2017-03-27T17:10:42] NOTE: accepted client connection
[2017-03-27T17:10:42] NOTE: client "xxxxx" has connected
[2017-03-27T17:10:47] NOTE: client "xxxxx" has disconnected
[2017-03-27T17:10:52] NOTE: accepted client connection
[2017-03-27T17:10:52] NOTE: client "xxxxx" has connected
```

Note the 5 second pauses each time.

**Comment by tsjk** [comment]
How come you have not managed to do something here in several years? I'm running a server using the latest snapshot (`synergy_1.14.6-snapshot.88fdd263_ubuntu20_amd64.deb`) on my linux system, and if I lock the server's desktop (which runs KDE Plasma v5.26.5, and uses Plasma's standard screen locking) I can no longer leave its screen. The server's screen seems to also be forced on, even though there is no activity there for some time.
Is there a work-around?

**Comment by desymless** [comment]
Something you can try/check:

1. Ensure that the `synergys` process is still running at the login window
2. Check if the network is available at the login screen. Not sure about the OS KDE Plasma, in other Linux distros (such as Ubuntu) it's possible to enable this [System->Preference->Network Connections. Choose the network name and select Edit. Select "Available to all users", and then click "Apply".]

**Comment by tsjk** [comment]
Thanks for your reply. I get the impression that your stance is that this should be working out of the box, despite the history of this issue. Is it supposed to?

So, some clarifications:
KDE Plasma is a desktop environment. I'm running the Gentoo distribution of Linux, on which I installed the .deb-package for Ubuntu v20 - which still depends on OpenSSL v1.1 (as I've yet not upgraded my system to use OpenSSL v3).
So, a possible parallel is the Kubuntu distribution, I suppose.

Anyway, `synergys` is running even after the screen is locked.

I'm not sure about your input on the topic of networking. Networking on my system is separated from the graphical interface, and of course everyone has access to the network.

In addition, this is related to screen locking, during which the login window (by which I suppose you mean the display manager) is not in play.

Some more details:
`synergys` is started when the display manager is started with a non-root user, using
`/usr/bin/synergys --debug INFO --name $HOSTNAME --address $IP_ADDRESS:24800 -c $HOME/.synergy/synergy.conf --enable-crypto --tls-cert $HOME/.synergy/SSL/Synergy.pem --no-tray --serial-key $SYNERGY_SERIAL_KEY`

When I lock the screen, it seems like the connection is severed, as the client claims the connection has been lost:
`[2023-02-01T11:12:45] WARNING: failed to connect to server: server is not responding`

Looking at the process list on the server, the `synergys` process is running, meaning that your code severs the connection for some reason.

Is this some XAuthorization problem? I noticed that I cannot actually use synergy until I login and get a session, even though `synergys` is spawned by the display manager. I'm not familiar with the technicalities of (Plasma) screen locking, so I don't know whether XAuthorization changes when the screen is locked.

**Comment by desymless** [comment]
My guess is that the network may not be available (for the processes) when the computer is locked, enabling the network connection for `all users` should allow the network to be available at the login screen.

But yes, it may be an authorization issue. A thought it to try to use Synergy 3 Beta and see if that helps. There is an option in the settings of the beta to "Allow Synergy to control this computer on the login screen".

**Comment by tsjk** [comment]
I can't really tell if you are serious or are just throwing random defensive arguments in salesman's role.
Could you give me an example of a mechanism that actually accomplishes what you claim is happening? That is, the disabling of network access for user processes when the screen is locked?

Anyway, just to make this argument go away I spawned a netcat listener  on the system running `synergys`, locked its screen, and then made a connection to that netcat listener from the client without a problem.

I might take a look at the beta. Or I might just try to get Logitech input switching to work between my systems. Some preliminary research have turned up some promising results.

Thanks again for your input.

**Comment by paulstelian97** [comment]
> I can't really tell if you are serious or are just throwing random defensive arguments in a salesman's role. Could you give me an example of a mechanism that actually accomplishes what you claim is happening? That is, the disabling of network access for user processes when the screen is locked?
> 

Not on Linux, but on macOS anything like VPNs, PPPoE connections and related get reset when you switch users, as such connections are only available per-user (unrelated, the same applies to Bluetooth). I believe something similar can happen on Windows as well, at least for the PPPoE connections. Simple LAN connections (plug in and it configures automatically) generally don't have such issues on any OS.

So yes, it's a legitimate connection.
